src/main/java/dude/component/Gui.java

```diff
@@ -0,0 +1,65 @@
+package dude.component;
+
+/**
+ * A class to handle dependencies/ownership between Duke which contains our application logic,
+ * and the JavaFX Application/actual GUI components.
+ * Duke has ownership of the Gui class, which holds reference to MainWindow which exposes methods to read input
+ * and display output but does not own it (MainWindow is owned by JavaFX Application).
+ * Duke and MainWindow thus contain references to each other without owning each other.
+ */
+public class Gui implements IUserInterface {
+    private MainWindow mainWindow;
+    private StringBuilder responseBuilder;
+
+    /**
+     * Initializes the Gui component of the Duke class that handles communication with the actual GUI from
+     * within the Duke class.
+     * Greets the user.
+     *
+     * @param mainWindow a reference to the MainWindow controller class so Gui can perform input/output by calling
+     *                   the public methods exposed by the controller class.
```

Comment:
Minor issue, but I'm not sure if the extra indentation is necessary; personally I feel it looks a little awkward.